### PR TITLE
Ws 5.6 approve sightings

### PR DESCRIPTION
**Pull request by natashabuckham**
- add new endpoint to sightings controller which gets all unapproved sightings
- add error handling
- add new method to the sightings service to support this
- write new fetch request in backendClient.ts to call this endpoint - import the header from the login context - passed as a prop from the admin page component
- set up a useEffect in Admin page which runs the fetch request on first render of the component
- set up a useState to store the sightings
- write the TSX to display the sightings - iterate through list from useState variable
- Add ‘approve’ button
- Add ‘approve’ fetch request to backendClient.ts
- link the button to an onClick which will trigger a fetch request to the admin controller ‘approve’ endpoint
- trigger a re-render of the page which will display the new list of sightings
- Add styling

### CODE REVIEW

**Comment by natashabuckham** [comment]
Right now the sightings response just gives you the User Id and Species Id, which we are displaying. We thought it could be another ticket to add more endpoints which get username by userid and species name by speciesid so that these can be displayed instead?

File: frontend/src/pages/Admin/Admin.tsx [hunk 1]
```diff
@@ -1,13 +1,76 @@
...
 const Admin = () => {
-  const { roleType } = useContext(LoginContext)
+  const loginContext = useContext(LoginContext)
+  const [sightings, setSightings] = useState<Sightings | null>(null)
+  const [approvals, setApprovals] = useState(0)
 
-  if (roleType === "Admin") {
+  useEffect(() => {
+    fetchUnapprovedSightings(loginContext.jwt).then((response) => setSightings(response))
+  }, [loginContext.jwt, approvals])
```
Comment (Kiran-R-K) [inline frontend/src/pages/Admin/Admin.tsx:13]: Why is `loginContext.jwt` in the useeffect? are we expecting it to be updated?

File: frontend/package.json [hunk 2]
```diff
@@ -11,6 +11,7 @@
     "@types/react": "^18.3.5",
     "@types/react-dom": "^18.3.0",
     "bootstrap": "^5.3.3",
+    "moment": "^2.30.1",
```
Comment (Kiran-R-K) [inline frontend/package.json:14]: Did you add this on purpose? what does it do?

**Review by Kiran-R-K — COMMENTED** [review]
looking really good, I've left you a couple of questions

**Comment by Kiran-R-K** [comment]
> Right now the sightings response just gives you the User Id and Species Id, which we are displaying. We thought it could be another ticket to add a new response model which passes the User and the Species objects, which then we can get the username by sighting.User.UserName and species name by sighting.Species.SpeciesName so that these can be displayed instead?

Can you add a ticket to cover this change for when the ticket working on the User and Species join to Sightings have been added

File: frontend/src/pages/Admin/Admin.tsx (hunk 1 above)
Comment (natashabuckham) [reply frontend/src/pages/Admin/Admin.tsx:13]: It won't pass the pre-commit checks without having that in there

File: frontend/package.json (hunk 2 above)
Comment (natashabuckham) [reply frontend/package.json:14]: We added it to parse the date, but ended up doing this another way, we will remove it

**Comment by natashabuckham** [comment]
> > Right now the sightings response just gives you the User Id and Species Id, which we are displaying. We thought it could be another ticket to add a new response model which passes the User and the Species objects, which then we can get the username by sighting.User.UserName and species name by sighting.Species.SpeciesName so that these can be displayed instead?
> 
> Can you add a ticket to cover this change for when the ticket working on the User and Species join to Sightings have been added

This has been sorted in Claudia and Ruksan's ticket now

**Comment by dorothyynwong** [comment]
Claudia and Ruksan's ticket handle displaying sightings for users only, I've raised a ticket for admin approve sightings page

File: frontend/src/pages/Admin/Admin.tsx (hunk 1 above)
Comment (Kiran-R-K) [reply frontend/src/pages/Admin/Admin.tsx:13]: strange but fair enough

**Review by Kiran-R-K — APPROVED** [review]

**Pull request merged**